### ISSUE TITLE
Improve error message 

### ISSUE THREAD

**Issue by vijaykiran** [sodadata/soda-core#1211]
```
Invalid check "columns types": no viable alternative at input 'columnstypes'
  +-> checks.yml line=4,col=5 "checks for actor"."[1]"(columns types)
Couldn't detect table check type of 'columns types'
  +-> checks.yml line=4,col=5 "checks for actor"."[1]"(columns types)
INFO   | Reading SodaCL file "environments.yml"
INFO   | Reading SodaCL file "checks.yml"
ERROR  | Invalid check "columns types": no viable alternative at input 'columnstypes' checks.yml line=4,col=5 "checks for actor"."[1]"(columns types)
ERROR  | Couldn't detect table check type of 'columns types' checks.yml line=4,col=5 "checks for actor"."[1]"(columns types)
```

- In the error "columnstypes" without spaces shows up
- The error is caught in 2 places ?

**Comment by tombaeyens** [comment]
Not sure if there is a simple fix for this.  I'm think the parsing has to get a serious rework.

Atm, we delegate merge the parsing of the check line and then the parsing of the optional check details.  When the check line is parsed, the antlr parser is straight connected to the error message logs.  

I think we should decouple that.  The Antlr parser logs are way worse than I expected.  I think we should just do the whole antlr parsing as a single operation that fails or succeeds.  And then just produce 1 error log saying that the check line is invalid without further detail.

Other thoughts on how we could fix this more simple?

**Comment by tombaeyens** [comment]
Merged into #1234
